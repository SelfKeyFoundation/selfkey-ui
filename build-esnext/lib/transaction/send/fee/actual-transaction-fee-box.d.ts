import * as React from 'react';
export declare const styles: Record<"root", import("@material-ui/core/styles/withStyles").CSSProperties | import("@material-ui/core/styles/withStyles").CreateCSSProperties<{}> | ((props: {}) => import("@material-ui/core/styles/withStyles").CreateCSSProperties<{}>)>;
export declare type ActualTransactionFeeBoxProps = {
    locale: string;
    ethFee: number;
    usdFee: number;
    fiatCurrency: string;
};
export declare const ActualTransactionFeeBox: React.ComponentType<Pick<ActualTransactionFeeBoxProps & {
    classes: Record<"root", string>;
}, "locale" | "fiatCurrency" | "ethFee" | "usdFee"> & import("@material-ui/core").StyledComponentProps<"root">>;
export default ActualTransactionFeeBox;
