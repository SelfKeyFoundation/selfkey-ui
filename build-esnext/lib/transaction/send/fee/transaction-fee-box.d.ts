import * as React from 'react';
import { WithStyles } from '@material-ui/core';
export declare const styles: Record<"checkbox" | "icon" | "fullWidth" | "container" | "formControl" | "formGroup" | "networkTransactionFeeTitle" | "showAdvancedContainer" | "rightIcon" | "downIcon" | "inputsContainer" | "checkboxRoot" | "checkboxChecked" | "checkboxLabel" | "nonceValue" | "currNetworkStatusContainer" | "currNetworkStatusTitle" | "reloadNetworkIcon" | "gasPriceValuesContainer", import("@material-ui/core/styles/withStyles").CSSProperties | import("@material-ui/core/styles/withStyles").CreateCSSProperties<{}> | ((props: {}) => import("@material-ui/core/styles/withStyles").CreateCSSProperties<{}>)>;
export declare type EthGasStationInfo = {
    safeLow: string;
    average: string;
    fast: string;
};
export declare type TransactionFeeBoxProps = {
    locale: string;
    fiatCurrency: string;
    ethFee: number;
    usdFee: number;
    gasLimit: number;
    gasPrice: number;
    nonce: number;
    showAdvanced?: boolean;
    ethGasStationInfo: EthGasStationInfo;
    reloadEthGasStationInfoAction?: ((event: React.MouseEvent<SVGSVGElement>) => void);
    changeGasLimitAction?: Function;
    changeGasPriceAction?: Function;
};
export declare type TransactionFeeBoxState = {
    showAdvanced: boolean;
    gasLimit: number;
    gasPrice: number;
};
export declare type StyledProps = WithStyles<typeof styles> & TransactionFeeBoxProps;
export declare class TransactionFeeBoxComponent extends React.Component<StyledProps, TransactionFeeBoxState> {
    timerToUpdateGasPrice: number;
    timerToUpdateGasLimit: number;
    TIME_FOR_INPUT_CHANGE: number;
    constructor(props: StyledProps);
    componentDidUpdate(prevProps: StyledProps): void;
    renderActualTransactionFeeBox(): JSX.Element;
    toggleShowAdvanced(): void;
    setGasLimit(event: React.ChangeEvent<HTMLInputElement>): void;
    setGasPrice(event: React.ChangeEvent<HTMLInputElement>): void;
    renderAdvancedContent(): JSX.Element;
    render(): JSX.Element;
}
export declare const TransactionFeeBox: React.ComponentType<Pick<StyledProps, "locale" | "nonce" | "fiatCurrency" | "ethFee" | "usdFee" | "gasLimit" | "gasPrice" | "showAdvanced" | "ethGasStationInfo" | "reloadEthGasStationInfoAction" | "changeGasLimitAction" | "changeGasPriceAction"> & import("@material-ui/core").StyledComponentProps<"checkbox" | "icon" | "fullWidth" | "container" | "formControl" | "formGroup" | "networkTransactionFeeTitle" | "showAdvancedContainer" | "rightIcon" | "downIcon" | "inputsContainer" | "checkboxRoot" | "checkboxChecked" | "checkboxLabel" | "nonceValue" | "currNetworkStatusContainer" | "currNetworkStatusTitle" | "reloadNetworkIcon" | "gasPriceValuesContainer">>;
export default TransactionFeeBox;
