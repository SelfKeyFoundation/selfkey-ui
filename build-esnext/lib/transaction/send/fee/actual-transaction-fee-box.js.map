{"version":3,"file":"actual-transaction-fee-box.js","sourceRoot":"","sources":["../../../../../src/transaction/send/fee/actual-transaction-fee-box.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAc,UAAU,EAAE,YAAY,EAAE,MAAM,mBAAmB,CAAC;AACzE,OAAO,WAAW,MAAM,8BAA8B,CAAC;AACvD,OAAO,EAAE,YAAY,EAAE,MAAM,8BAA8B,CAAC;AAC5D,OAAO,EAAE,IAAI,EAAE,MAAM,mBAAmB,CAAC;AAEzC,MAAM,CAAC,MAAM,MAAM,GAAG,YAAY,CAAC;IAC/B,IAAI,EAAE;QACF,KAAK,EAAE,SAAS;QAChB,QAAQ,EAAE,MAAM;QAChB,UAAU,EAAE,WAAW,CAAC,UAAU;KACrC;CACJ,CAAC,CAAC;AASH,MAAM,CAAC,MAAM,uBAAuB,GAAG,UAAU,CAAC,MAAM,CAAC,CACrD,CAAC,EAAC,OAAO,EAAE,MAAM,EAAE,MAAM,EAAE,MAAM,EAAE,YAAY,EAA4D,EAAE,EAAE,CAAC,CAC5G,oBAAC,IAAI,IAAC,SAAS,QAAC,SAAS,EAAE,OAAO,CAAC,IAAI,EAAE,SAAS,EAAC,KAAK,EAAC,OAAO,EAAE,CAAC;IAC/D,oBAAC,IAAI,IAAC,IAAI;QACN,oBAAC,IAAI,IAAC,SAAS,QAAC,OAAO,EAAE,CAAC;YACtB,oBAAC,IAAI,IAAC,IAAI;gBACN,oBAAC,YAAY,IAAC,MAAM,EAAE,MAAM,EAAE,UAAU,EAAC,SAAS,EAAC,QAAQ,EAAC,KAAK,EAAC,KAAK,EAAE,MAAM,EAAE,cAAc,EAAE,EAAE,GAAG,CACnG;YACP,oBAAC,IAAI,IAAC,IAAI,gBAEH;YACP,oBAAC,IAAI,IAAC,IAAI,cAEH,CACJ,CACJ;IACP,oBAAC,IAAI,IAAC,IAAI;QACN,oBAAC,IAAI,IAAC,SAAS,QAAC,OAAO,EAAE,CAAC;YACtB,oBAAC,IAAI,IAAC,IAAI;gBACN,oBAAC,YAAY,IAAC,MAAM,EAAE,MAAM,EAAE,UAAU,EAAC,UAAU,EAAC,QAAQ,EAAE,YAAY,EAAE,KAAK,EAAE,MAAM,EAAE,cAAc,EAAE,EAAE,GAAG,CAC7G;YACP,oBAAC,IAAI,IAAC,IAAI,UACL,YAAY,CACV,CACJ,CACJ,CACJ,CACd,CAAC,CAAC;AAEH,eAAe,uBAAuB,CAAC"}