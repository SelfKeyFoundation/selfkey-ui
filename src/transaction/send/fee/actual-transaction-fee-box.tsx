import * as React from 'react';
// @ts-ignore
import injectSheet, { StyleSheet } from 'react-jss';
import CommonStyle from '../../../common/common-style';
import { NumberFormat } from '../../../price/number-format';
import { Grid } from '@material-ui/core';

export const styles: StyleSheet = {
    root: {
        color: '#FFFFFF',
        fontSize: '16px',
        fontFamily: CommonStyle.fontFamily,
    }
};

export type ActualTransactionFeeBoxProps = {
    locale: string,
    ethFee: number,
    usdFee: number,
    fiatCurrency: string
}

export const ActualTransactionFeeBox = injectSheet(styles)<ActualTransactionFeeBoxProps>(({classes, locale, ethFee, usdFee, fiatCurrency }) => (
    <Grid container className={classes.root} direction='row' spacing={8}>
        <Grid item>
            <Grid container spacing={8}>
                <Grid item>
                    <NumberFormat locale={locale} priceStyle='decimal' currency='ETH' value={ethFee} fractionDigits={15}/>
                </Grid>
                <Grid item>
                    ETH
                </Grid>
                <Grid item>
                    /
                </Grid>
            </Grid>
        </Grid>
        <Grid item>
            <Grid container spacing={8}>
                <Grid item>
                    <NumberFormat locale={locale} priceStyle='currency' currency={fiatCurrency} value={usdFee} fractionDigits={15}/>
                </Grid>
                <Grid item>
                    {fiatCurrency}
                </Grid>
            </Grid>
        </Grid>
    </Grid>
));

export default ActualTransactionFeeBox;
