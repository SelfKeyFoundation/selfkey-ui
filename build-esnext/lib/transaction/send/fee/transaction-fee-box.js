import * as React from 'react';
import { Grid } from '@material-ui/core';
import { withStyles, createStyles } from '@material-ui/styles';
import { ActualTransactionFeeBox } from './actual-transaction-fee-box';
import CommonStyle from '../../../common/common-style';
import { Loop as LoopIcon } from '@material-ui/icons';
export const styles = createStyles({
    container: {
        fontFamily: CommonStyle.fontFamily
    },
    networkTransactionFeeTitle: {
        color: '#93B0C1',
        fontSize: '16px',
        paddingRight: '5px'
    },
    showAdvancedContainer: {
        color: '#00C0D9',
        cursor: 'pointer',
        fontSize: '14px',
        minWidth: '85px',
        textAlign: 'right'
    },
    icon: {
        border: 'solid #00C0D9',
        borderWidth: '0 2px 2px 0',
        display: 'inline-block',
        padding: '4px',
        marginLeft: '5px'
    },
    rightIcon: {
        transform: 'rotate(-45deg)'
    },
    downIcon: {
        transform: 'rotate(45deg)'
    },
    inputsContainer: {
        paddingTop: '52px'
    },
    formGroup: {
        display: 'flex',
        flexDirection: 'column',
        '&& label': {
            fontSize: '12px',
            fontWeight: 600,
            marginBottom: '10px',
            lineHeight: '15px',
            color: '#93A4AF'
        }
    },
    fullWidth: {
        width: '100%'
    },
    checkbox: {
        color: '#FFFFFF',
        fontSize: '14px',
        lineHeight: '14px'
    },
    checkboxRoot: {
        color: '#00C0D9',
        '&$checked': {
            color: '#00C0D9 !important'
        }
    },
    checkboxChecked: {
        color: '#00C0D9 !important'
    },
    checkboxLabel: {
        color: '#FFFFFF',
        fontSize: '14px',
        lineHeight: '14px'
    },
    nonceValue: {
        color: '#FFFFFF',
        fontSize: '14px',
        lineHeight: '14px'
    },
    formControl: {
        paddingLeft: '12px',
        boxSizing: 'border-box',
        height: '46px',
        width: '178px',
        border: '1px solid #384656',
        borderRadius: '4px',
        backgroundColor: '#1E262E',
        boxShadow: 'inset -1px 0 0 0 rgba(0,0,0,0.24), 1px 0 0 0 rgba(118,128,147,0.2), 2px 0 2px 0 rgba(0,0,0,0.2)',
        color: '#FFFFFF',
        fontSize: '14px',
        lineHeight: '14px',
        '&:focus': {
            outline: 'none',
            boxShadow: '0 0 5px rgba(81, 203, 238, 1)'
        }
    },
    currNetworkStatusContainer: {
        display: 'flex',
        paddingTop: '9px'
    },
    currNetworkStatusTitle: {
        color: '#93B0C1',
        fontSize: '13px',
        lineHeight: '19px',
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'column'
    },
    reloadNetworkIcon: {
        color: '#93B0C1',
        cursor: 'pointer'
    },
    gasPriceValuesContainer: {
        paddingTop: '15px;',
        '&& p': {
            color: '#FFFFFF',
            fontSize: '14px',
            lineHeight: '26px',
            margin: 0
        }
    }
});
export class TransactionFeeBoxComponent extends React.Component {
    constructor(props) {
        super(props);
        this.timerToUpdateGasPrice = 0;
        this.timerToUpdateGasLimit = 0;
        this.TIME_FOR_INPUT_CHANGE = 1000;
        this.state = {
            showAdvanced: props.showAdvanced || false,
            gasLimit: props.gasLimit,
            gasPrice: props.gasPrice
        };
    }
    componentDidUpdate(prevProps) {
        if (prevProps.gasLimit !== this.props.gasLimit ||
            prevProps.gasPrice !== this.props.gasPrice) {
            this.setState({
                ...this.state,
                gasLimit: this.props.gasLimit,
                gasPrice: this.props.gasPrice
            });
        }
    }
    renderActualTransactionFeeBox() {
        return React.createElement(ActualTransactionFeeBox, Object.assign({}, this.props));
    }
    toggleShowAdvanced() {
        const { showAdvanced } = this.state;
        this.setState({ ...this.state, showAdvanced: !showAdvanced });
    }
    setGasLimit(event) {
        const value = event.target.value;
        if (this.timerToUpdateGasLimit)
            clearTimeout(this.timerToUpdateGasLimit);
        this.setState({ ...this.state, gasLimit: Number(value) });
        this.timerToUpdateGasLimit = window.setTimeout(() => {
            if (this.props.changeGasLimitAction) {
                this.props.changeGasLimitAction(value);
            }
        }, this.TIME_FOR_INPUT_CHANGE);
    }
    setGasPrice(event) {
        const value = event.target.value;
        if (this.timerToUpdateGasPrice)
            clearTimeout(this.timerToUpdateGasPrice);
        this.setState({ ...this.state, gasPrice: Number(value) });
        this.timerToUpdateGasPrice = window.setTimeout(() => {
            if (this.props.changeGasPriceAction) {
                this.props.changeGasPriceAction(value);
            }
        }, this.TIME_FOR_INPUT_CHANGE);
    }
    renderAdvancedContent() {
        const { classes, ethGasStationInfo, reloadEthGasStationInfoAction, nonce } = this.props;
        return (React.createElement("div", { className: classes.fullWidth },
            React.createElement(Grid, { container: true, className: classes.inputsContainer, direction: "row", justify: "space-between", alignItems: "flex-start" },
                React.createElement("div", { className: classes.formGroup },
                    React.createElement("label", null, "Gas Price (Gwei)"),
                    React.createElement("input", { type: "text", className: classes.formControl, value: this.state.gasPrice, onChange: e => this.setGasPrice(e) })),
                React.createElement("div", null,
                    React.createElement("div", { className: classes.formGroup },
                        React.createElement("label", null, "Gas Limit"),
                        React.createElement("input", { type: "text", value: this.state.gasLimit, onChange: e => this.setGasLimit(e), className: classes.formControl }))),
                React.createElement("div", { className: classes.formGroup },
                    React.createElement("label", null, "Nonce"),
                    React.createElement("span", { className: classes.nonceValue },
                        " ",
                        nonce,
                        " "))),
            React.createElement(Grid, { container: true, direction: "column", justify: "center", alignItems: "center" },
                React.createElement("div", { className: classes.currNetworkStatusContainer },
                    React.createElement("span", { className: classes.currNetworkStatusTitle },
                        "Current Network Status:",
                        ' '),
                    React.createElement(LoopIcon, { onClick: reloadEthGasStationInfoAction, classes: {
                            root: classes.reloadNetworkIcon
                        } }, ' ')),
                React.createElement("div", { className: classes.gasPriceValuesContainer },
                    React.createElement("p", null,
                        " Under 30 Mins: ",
                        ethGasStationInfo.safeLow,
                        " Gwei "),
                    React.createElement("p", null,
                        " Under 5 Mins: ",
                        ethGasStationInfo.average,
                        " Gwei "),
                    React.createElement("p", null,
                        " Under 2 Mins: ",
                        ethGasStationInfo.fast,
                        " Gwei ")))));
    }
    render() {
        const { classes } = this.props;
        const { showAdvanced } = this.state;
        return (React.createElement(Grid, { container: true, direction: "row", justify: "space-between", alignItems: "center", className: classes.container },
            React.createElement(Grid, { container: true, wrap: "nowrap" },
                React.createElement(Grid, { container: true, direction: "row" },
                    React.createElement(Grid, { item: true },
                        React.createElement("span", { className: `${classes.networkTransactionFeeTitle} feeTitle` },
                            ' ',
                            "Network Transaction Fee:",
                            ' ')),
                    React.createElement(Grid, { item: true }, this.renderActualTransactionFeeBox())),
                React.createElement(Grid, { item: true, className: classes.showAdvancedContainer, onClick: () => this.toggleShowAdvanced() },
                    React.createElement("span", null, " Advanced "),
                    !showAdvanced ? (React.createElement("i", { className: `${classes.icon}  ${classes.rightIcon}` }, " ")) : (React.createElement("i", { className: `${classes.icon}  ${classes.downIcon}` }, " ")))),
            showAdvanced && this.renderAdvancedContent()));
    }
}
export const TransactionFeeBox = withStyles(styles)(TransactionFeeBoxComponent);
export default TransactionFeeBox;
//# sourceMappingURL=transaction-fee-box.js.map