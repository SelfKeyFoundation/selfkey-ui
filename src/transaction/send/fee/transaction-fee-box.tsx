import * as React from 'react';
// @ts-ignore
import injectSheet, { WithStyles, StyleSheet, StyledComponentProps, ClassNameMap } from 'react-jss';
import { ActualTransactionFeeBox } from './actual-transaction-fee-box';
import CommonStyle from '../../../common/common-style';
import { Grid } from '@material-ui/core';
import { Loop as LoopIcon } from '@material-ui/icons';

export const styles: StyleSheet = {
    container: {
        fontFamily: CommonStyle.fontFamily
    },
    networkTransactionFeeTitle: {
        paddingRight: '5px',
        color: '#93B0C1',
        fontSize: '16px'
    },
    showAdvancedContainer: {
        cursor: 'pointer',
        fontSize: '14px',
        color: '#00C0D9'
    },

    icon: {
        border: 'solid #00C0D9',
        borderWidth: '0 2px 2px 0',
        display: 'inline-block',
        padding: '4px',
        marginLeft: '5px',
    },
    rightIcon: {
        transform: 'rotate(-45deg)',
        '-webkitTransform': 'rotate(-45deg)'
    },
    downIcon: {
        transform: 'rotate(45deg)',
        '-webkit-transform': 'rotate(45deg)'
    },
    inputsContainer: {
        paddingTop: '52px'
    },
    formGroup: {
        display: 'flex',
        flexDirection: 'column',
        '&& label': {
            fontSize: '12px',
            fontWeight: '600',
            marginBottom: '10px',
            lineHeight: '15px',
            color: '#93A4AF'
        }
    },
    fullWidth: {
        width: '100%',
    },
    checkbox: {
        color: '#FFFFFF', fontSize: '14px', lineHeight: '14px'
    },
    checkboxRoot: {
        color: '#00C0D9',
        '&$checked': {
            color: '#00C0D9 !important'
        }
    },
    checkboxChecked: {
        color: '#00C0D9 !important',
    },

    checkboxLabel: {
        color: '#FFFFFF',
        fontSize: '14px',
        lineHeight: '14px'
    },
    nonceValue: {
        color: '#FFFFFF',
        fontSize: '14px',
        lineHeight: '14px'
    },
    formControl: {
        paddingLeft: '12px',
        boxSizing: 'border-box',
        height: '46px',
        width: '178px',
        border: '1px solid #384656',
        borderRadius: '4px',
        backgroundColor: '#1E262E',
        boxShadow: 'inset -1px 0 0 0 rgba(0,0,0,0.24), 1px 0 0 0 rgba(118,128,147,0.2), 2px 0 2px 0 rgba(0,0,0,0.2)',
        color: '#FFFFFF', fontSize: '14px', lineHeight: '14px',
        '&:focus': {
            outline: 'none',
            boxShadow: '0 0 5px rgba(81, 203, 238, 1)'
        }
    },
    currNetworkStatusContainer: {
        display: 'flex',
        paddingTop: '9px'
    },
    currNetworkStatusTitle: {
        color: '#93B0C1',
        fontSize: '13px',
        lineHeight: '19px',
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'column'
    },
    reloadNetworkIcon: {
        color: '#93B0C1',
        cursor: 'pointer'
    },
    gasPriceValuesContainer: {
        paddingTop: '15px;',
        '&& p': {
            color: '#FFFFFF',
            fontSize: '14px',
            lineHeight: '26px',
            margin: 0
        }
    }
};

export type EthGasStationInfo = {
    safeLow: string,
    average: string,
    fast: string
};

export type TransactionFeeBoxProps = {
    locale: string,
    fiatCurrency: string,
    ethFee: number,
    usdFee: number,
    gasLimit: number,
    gasPrice: number
    nonce: number,
    showAdvanced?: boolean,
    ethGasStationInfo: EthGasStationInfo,
    reloadEthGasStationInfoAction?: ((event: React.MouseEvent<SVGSVGElement>) => void)
    changeGasLimitAction?: Function,
    changeGasPriceAction?: Function
}

export type TransactionFeeBoxState = {
    showAdvanced: boolean
    gasLimit: number,
    gasPrice: number
};

export type StyledProps = WithStyles<keyof typeof styles> & TransactionFeeBoxProps;

export class TransactionFeeBoxComponent extends React.Component<StyledProps, TransactionFeeBoxState> {
    timerToUpdateGasPrice = 0;
    timerToUpdateGasLimit = 0;

    TIME_FOR_INPUT_CHANGE = 1000;

    constructor(props: StyledProps) {
        super(props);

        this.state = {
            showAdvanced: props.showAdvanced || false,
            gasLimit: props.gasLimit,
            gasPrice: props.gasPrice
        }
    }

    componentDidUpdate(prevProps: StyledProps) {
        if(prevProps.gasLimit !== this.props.gasLimit || prevProps.gasPrice !== this.props.gasPrice) {
            this.setState({ ...this.state, gasLimit: this.props.gasLimit, gasPrice: this.props.gasPrice });
        }
    }

    renderActualTransactionFeeBox() {
        return (
            <ActualTransactionFeeBox {...this.props}/>
        );
    }

    toggleShowAdvanced() {
        const { showAdvanced } = this.state;
        this.setState({ ...this.state, showAdvanced: !showAdvanced });
    }

    setGasLimit(event: React.ChangeEvent<HTMLInputElement>) {
        const value = event.target.value;
        if(this.timerToUpdateGasLimit) clearTimeout(this.timerToUpdateGasLimit);
        this.setState({ ...this.state, gasLimit: Number(value) });

        this.timerToUpdateGasLimit = window.setTimeout(()=>{
            if(this.props.changeGasLimitAction) {
                this.props.changeGasLimitAction(value);
            }
        }, this.TIME_FOR_INPUT_CHANGE);

    }

    setGasPrice(event: React.ChangeEvent<HTMLInputElement>) {
        const value = event.target.value;
        if(this.timerToUpdateGasPrice) clearTimeout(this.timerToUpdateGasPrice);
        this.setState({ ...this.state, gasPrice: Number(value) });

        this.timerToUpdateGasPrice = window.setTimeout(()=>{
            if(this.props.changeGasPriceAction) {
                this.props.changeGasPriceAction(value);
            }
        }, this.TIME_FOR_INPUT_CHANGE);
    }

    renderAdvancedContent() {
        const { classes, ethGasStationInfo, reloadEthGasStationInfoAction, nonce } = this.props;
        return (
            <div className={classes.fullWidth}>
                <Grid container className={classes.inputsContainer} direction="row" justify="space-between" alignItems="flex-start">
                    <div className={classes.formGroup}>
                        <label>Gas Price (Gwei)</label>
                        <input type="text" className={classes.formControl} value={this.state.gasPrice} onChange={(e) => this.setGasPrice(e)} />
                    </div>

                    <div>
                        <div className={classes.formGroup}>
                            <label>Gas Limit</label>
                            <input type="text" value={this.state.gasLimit} onChange={(e) => this.setGasLimit(e)} className={classes.formControl} />
                        </div>
                    </div>
                    <div className={classes.formGroup}>
                        <label>Nonce</label>
                        <span className={classes.nonceValue}> {nonce} </span>
                    </div>
                </Grid>

                <Grid container direction="column" justify="center" alignItems="center">
                    <div className={classes.currNetworkStatusContainer}>
                        <span className={classes.currNetworkStatusTitle}>Current Network Status: </span>
                        <LoopIcon onClick={reloadEthGasStationInfoAction} classes={{
                            root: classes.reloadNetworkIcon
                        }}> </LoopIcon>
                    </div>
                    <div className={classes.gasPriceValuesContainer}>
                        <p> Under 30 Mins: {ethGasStationInfo.safeLow} Gwei </p>
                        <p> Under 5 Mins: {ethGasStationInfo.average} Gwei </p>
                        <p> Under 2 Mins: {ethGasStationInfo.fast} Gwei </p>
                    </div>
                </Grid>
            </div>
        );
    }

    render() {
        let { classes } = this.props;
        let { showAdvanced } = this.state;
        return (
            <Grid container direction="row" justify="space-between" alignItems="center" className={classes.container}>
                <Grid item>
                    <Grid container direction="row">
                        <Grid item>
                            <span className={classes.networkTransactionFeeTitle}> Network Transaction Fee: </span>
                        </Grid>
                        <Grid item>
                            {this.renderActualTransactionFeeBox()}
                        </Grid>
                    </Grid>
                </Grid>
                <Grid item className={classes.showAdvancedContainer} onClick={() => this.toggleShowAdvanced()}>
                    <span> Advanced </span>
                    {!showAdvanced ? (
                        <i className={`${classes.icon}  ${classes.rightIcon}`}> </i>
                    ) : (
                            <i className={`${classes.icon}  ${classes.downIcon}`}> </i>
                        )}
                </Grid>
                {showAdvanced && this.renderAdvancedContent()}
            </Grid>
        );
    }
}

export const TransactionFeeBox = injectSheet(styles)(TransactionFeeBoxComponent);

export default TransactionFeeBox;
