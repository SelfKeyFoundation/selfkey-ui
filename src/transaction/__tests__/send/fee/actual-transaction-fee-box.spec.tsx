import * as React from 'react';
import { create } from 'react-test-renderer';
import { ActualTransactionFeeBox } from '../../../send/fee/actual-transaction-fee-box';

describe('Transaction Fee', () => {
	it('should show Transaction Fee screen', () => {
		const component = create(
			<ActualTransactionFeeBox locale="en" ethFee={0.0001} usdFee={0.02} fiatCurrency="USD" />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
