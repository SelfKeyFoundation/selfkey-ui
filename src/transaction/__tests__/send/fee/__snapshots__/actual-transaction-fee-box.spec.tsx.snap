// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Fee should show Transaction Fee screen 1`] = `
<div
  className="MuiGrid-root Component-root-0-2-1 MuiGrid-container MuiGrid-spacing-xs-1"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div>
          0.0001
           
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        ETH
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        /
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div>
          $0.02
           
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        USD
      </div>
    </div>
  </div>
</div>
`;
